hg-fast-export.py

```diff
@@ -157,7 +176,7 @@ def refresh_git_submodule(name,subrepo_info):
   return b'[submodule "%s"]\n\tpath = %s\n\turl = %s\n' % (name, name, subrepo_info[0])
 
 def refresh_hg_submodule(name,subrepo_info):
-  gitRepoLocation=submodule_mappings[name] + b"/.git"
+  gitRepoLocation=submodule_mappings[name]['path'] + b"/.git"
```

Comment:
How will this interact with an existing mappings file for an incremental import?

Reply:
This has been used to convert many remote repositories in bitbucket. The mapping file is exactly the same as now. However, the 'path' points to the local copy of the repo and the 'url' points to the remote. 'path' is used to convert without change. However, when the .gitmodules file is written the 'url' is used. 'url' will be same as 'path' if there is no remote.

Reply:
I took another look, this will probably fly for your example, but changing `load_mapping()` to instead of mapping strings to strings, map strings to dicts instead will pretty much break all other uses of mappings in fast-export (grep for load_mapping, to find all call sites).